src/Server/CertReloader.cpp

```diff
@@ -0,0 +1,99 @@
+#include "CertReloader.h"
+
+namespace DB
+{
+#if USE_SSL
+int cert_reloader_dispatch_set_cert(SSL * ssl, [[maybe_unused]] void * arg)
+{
+    return CertReloader::instance().SetCert(ssl);
+}
+
+int CertReloader::SetCert(SSL * ssl)
+{
+    std::shared_lock lock(lck);
+    SSL_use_certificate(ssl, const_cast<X509 *>(cert->certificate()));
```

Comment:
You have to load in the intermediates as well, and that's apparently a different function, not documented in the manpages.  I don't know if just changing the end cert lets the intermediates work.

Reply:
Also I'm not sure if SSL_user_certificate transfers ownership or not.

Reply:
> You have to load in the intermediates as well, and that's apparently a different function, not documented in the manpages. I don't know if just changing the end cert lets the intermediates work.

I did a bit of testing and it seems that providing a correcly configured ca path does the trick xD